src/Components/Components/src/NavigationOptions.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Additional options for navigating to another URI.
+    /// </summary>
+    public readonly struct NavigationOptions
+    {
+        /// <summary>
+        /// If true, bypasses client-side routing and forces the browser to load the new page from the server, whether or not the URI would normally be handled by the client-side router.
+        /// </summary>
+        public bool ForceLoad { get; init; }
+
+        /// <summary>
+        /// If true, replaces the currently entry in the history stack.
+        /// If false, appends the new entry to the history stack.
+        /// </summary>
+        public bool ReplaceHistoryEntry { get; init; }
+    }
```

Comment:
In an earlier edition of this PR, `NavigationOptions` was a flags enum. I've changed it to a full-fledged `struct` because I don't want to get into this situation again in the future where we need to add some other option and have to multiply out the number of overloads and obscure rules for what methods you have to implement in subclasses.

With a struct, even if we need new options of type `string`, or rules about which combinations of options are legal to use simultaneously, it should be possible to model that (e.g., as different constructor overloads).